library/src/main/java/kwasm/ast/FunctionType.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kwasm.ast
+
+/**
+ * Data class to hold the parameters and results of a FunctionType
+ * from [the docs](https://webassembly.github.io/spec/core/text/types.html#function-types):
+ *
+ * Where functype is a [kwasm.format.text.Type.FunctionType]:
+ *
+ * ```
+ *   functype ::=  ‘(’ ‘func’  t∗1:vec(param)  t∗2:vec(result) ‘)’ -> [t∗1]→[t∗2]
```

Comment:
Optional Ultra-Nit: I like to replace the subscript `*`s with normal `*`s when I copy the grammar over.

Reply:
Also, I like to use `=>` instead of `->`

Reply:
Done